chore: add analytics events for trace funnels

ahmadshaheer opened the pull request:
### Summary

<!-- ✍️ A clear and concise description...-->

#### Related Issues / PR's
close https://github.com/SigNoz/engineering-pod/issues/2305
<!-- ✍️ Add the issues being resolved here and related PR's where applicable  -->

#### Screenshots

NA

<!-- ✍️ Add screenshots of before and after changes where applicable-->

#### Affected Areas and Manually Tested Areas

<!-- ✍️ Add details of blast radius and dev testing areas where applicable-->

<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Add analytics events to log user interactions with trace funnels across various components.
> 
>   - **Analytics Events**:
>     - Add `logEvent` calls in `TraceDetailV2/index.tsx` to log visits to trace funnels from trace details page.
>     - Add `logEvent` in `StepsContent.tsx` to log when a span is added as a new step from trace details page.
>     - Add `logEvent` in `FunnelContext.tsx` to log when a span is replaced from trace details page.
>     - Add `logEvent` in `CreateFunnel.tsx` to log funnel creation from trace details or funnels list page.
>     - Add `logEvent` in `TracesModulePage.tsx` to log visits to trace funnels from trace explorer page.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for 9f5f0c0da221c4a6826038b13af1ccca02337200. You can [customize](https://app.ellipsis.dev/SigNoz/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

shivanshuraj1333 commented:
@ahmadshaheer I don't see all the events from https://www.notion.so/signoz/Analytics-Events-Doc-e43794709fc94dd19ff650b56806a5d2?pvs=4 added here

from the analytics events we should be able to create mix panel dashboard and understand how many people are using it, and how are they using it. We need to add the analytics in that way.

This is not a release blocker though.

shivanshuraj1333 reviewed (changes requested):
need to add events which can help answer some key questions, see the comment for more details

ahmadshaheer commented:
> @ahmadshaheer I don't see all the events from https://www.notion.so/signoz/Analytics-Events-Doc-e43794709fc94dd19ff650b56806a5d2?pvs=4 added here
> 
> from the analytics events we should be able to create mix panel dashboard and understand how many people are using it, and how are they using it. We need to add the analytics in that way.
> 
> This is not a release blocker though.

@shivanshuraj1333, I had initially shared around 9 events on this [thread](https://signoz-team.slack.com/archives/C089MNX4Y90/p1744707694455759?thread_ts=1744696364.798049&cid=C089MNX4Y90). in the doc, you had mentioned to remove the strike through ones. leaving just [3 items](https://www.notion.so/signoz/Analytics-Events-Doc-e43794709fc94dd19ff650b56806a5d2?pvs=4#1d6fcc6bcd19804dae3ec2e1366f258e), i.e.

> 3. Trace Funnels: Funnel created from trace details page
> 4. Trace Funnels: Funnel created from trace funnels list page
> 5. Trace Funnels: span added from trace details page

in this PR, I have added those events, except for 3rd (`Trace Funnels: Funnel created from trace details page`) item (I added 2 events i.e. 1 for adding span as a new step of a funnel, and 1 for replacing an existing step in the funnel).

ahmadshaheer commented:
@shivanshuraj1333, as discussed over the call, I'll add the first 2 events of the doc as well.

shivanshuraj1333 reviewed (dismissed):
please resolve the conflicts, thanks!

aks07 reviewed (dismissed):
Make sure changes from the other PR are not undone here.
